base/ca/src/org/dogtagpki/server/ca/rest/CertService.java

```diff
@@ -209,12 +217,30 @@ public Response revokeCert(CertId id, CertRevokeRequest request, boolean caCert)
             if (clientCert != null) {
                 clientSerialNumber = clientCert.getSerialNumber();
                 clientSubjectDN = clientCert.getSubjectDN().toString();
-                clientRecord = processor.getCertificateRecord(clientSerialNumber);
 
-                // Verify client cert is not revoked.
-                // TODO: This should be checked during authentication.
-                if (clientRecord.getStatus().equals(ICertRecord.STATUS_REVOKED)) {
-                    throw new UnauthorizedException(CMS.getLogMessage("CMSGW_UNAUTHORIZED"));
+                String clientIssuerDN = clientCert.getIssuerDN().toString();
```

Comment:
clientCert is X509Certificate in jss, which defines getissuerDN() as the following:
public abstract Principal getIssuerDN();
Yeah it could get confusing...